OPHD/Technology/TechnologyCatalog.cpp

```diff
@@ -166,21 +166,25 @@ namespace
 	}
 
 
-	std::map<std::string, std::vector<Technology>> readCategories(NAS2D::Xml::XmlElement& node)
+	std::vector<TechnologyCatalog::Category> readCategories(NAS2D::Xml::XmlElement& node)
 	{
-		std::map<std::string, std::vector<Technology>> categories;
+		std::vector<TechnologyCatalog::Category> categories;
 		for (auto category = &node; category; category = category->nextSiblingElement())
 		{
 			const auto attributes = NAS2D::attributesToDictionary(*category);
-			const auto name = attributes.get("name");
+			TechnologyCatalog::Category newCategory{std::stoi(attributes.get("icon_index")), attributes.get("name"), {}};
 
-			auto it = categories.find(name);
-			if (it != categories.end())
+			for (const auto& cat : categories)
 			{
-				throw std::runtime_error("TechnologyReader: Category redefinition '" + name + "'" + nodeAtString(*category));
+				if (cat.name == newCategory.name)
+				{
+					throw std::runtime_error("TechnologyReader: Category redefinition '" + newCategory.name + "'" + nodeAtString(*category));
+				}
 			}
-			categories[name] = readTechnologiesInCategory(*category);
+			newCategory.technologies = readTechnologiesInCategory(*category);

```

Comment:
Potentially, you could delay construction of `newCategory` to this point, which avoids the need to initialize the `std::vector<Technology>` and then overwriting it with a new `vector`. Though the category `name` is needed for the redefinition check, so if that is read ahead of time you'd need to `std::move` it into the `Catagory` to avoid copy/re-init.